### ISSUE TITLE
`AwaitExpression` does not include `ImportExpression` as possible value of `argument` property

### ISSUE THREAD

**Issue by G-Rath** [typescript-eslint/typescript-eslint#4877]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgQQMoBUD6A5A8gEQFFN0BNABUNQBo51Ur0oBTZuAXzgDMoIQ4A5AAEYATzDMAzgGMowMDAC0UgDbAAdjAD0AVxjAVkgQG4AsACgL0iOsnwAJsxjMoIDcwCS4aDFQQdUNJsALxwABQWcHDqEI4AXHQMGCzMAHTIAO4AhsAwhAAeYCySksA21BYAlAl2cuoA5nAAPtE6KipwwQB8iJFwwFzhMY6pWVD1OiDMmqliEp3BoWhYeEQkFFSpXpCwBUVSpTaVveZRUVpafewWfSwwAeqt7Wbm7MZAA

```ts
import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/utils';

const determineImportSource = (
  node: TSESTree.AwaitExpression,
): string | null => {
  if (node.argument.type === AST_NODE_TYPES.ImportExpression) {
    //
  }

  return null;
};
```

**Expected Result**

No errors

**Actual Result**

An error from TypeScript:

```
This condition will always return 'false' since the types 'AST_NODE_TYPES.ArrayExpression | AST_NODE_TYPES.ArrayPattern | AST_NODE_TYPES.ArrowFunctionExpression | AST_NODE_TYPES.AwaitExpression | AST_NODE_TYPES.CallExpression | AST_NODE_TYPES.ClassExpression | ... 18 more ... | AST_NODE_TYPES.TSTypeAssertion' and 'AST_NODE_TYPES.ImportExpression' have no overlap
```

**Additional Info**

Here's an AST tree showing that it's possible to have an import expression as the argument of an await expression:

```
await import('@jest/globals');
```

![image](https://user-images.githubusercontent.com/3151613/165879207-4595bbdf-b02f-43ef-8a2d-72c3ecae331e.png)

https://astexplorer.net/#/gist/bd61e9f9222ff844ddbd1748895d8f8f/07c5c90e2933a965ef5329f4eb68588910343271

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/types` | `5.20.0` |
| `TypeScript`                           | `4.6.3` |
| `node`                                 | `v14.18.2` |

**Comment by G-Rath** [comment]
(I don't think `typescript-estree` is actually the package this bug is in - sorry 😅)

**Comment by bradzacher** [comment]
`ImportExpression` should be added to [`LeftHandSideExpression`](https://github.com/typescript-eslint/typescript-eslint/blob/f79ae9b58e82f4fddef640a34a1d7ff92b763e65/packages/ast-spec/src/unions/LeftHandSideExpression.ts#L23).
